linux_os/guide/system/software/integrity/fips/enable_fips_mode/ansible/shared.yml

```diff
@@ -4,5 +4,12 @@
 # complexity = medium
 # disruption = medium
 
-- name: enable fips mode
+- name: Check to see the current status of FIPS mode
+  command: /usr/bin/fips-mode-setup --check
+  register: is_fips_enabled
+  changed_when: false
+
+- name: Enable FIPS mode
   command: /usr/bin/fips-mode-setup --enable
+  when:
+    - is_fips_enabled.stdout.find('FIPS mode is enabled.') == -1
```

Comment:
Nitpick: 
Should we also check that there is only one line in the output?
So that when the crypto-policy is not FIPS or FIPS with modules it resets the crypto-policy to FIPS?
It may not matter much as the rule is usually paired with `configure_crypto_policy`.

Reply:
I think its fine since it goes along perfectly with our OCIL check content.